codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Headers.java

```diff
@@ -35,51 +46,86 @@
         /**
          * {@code :method}.
          */
-        METHOD(":method"),
+        METHOD(":method", REQUEST_PSEUDO_HEADER),
 
         /**
          * {@code :scheme}.
          */
-        SCHEME(":scheme"),
+        SCHEME(":scheme", REQUEST_PSEUDO_HEADER),
 
         /**
          * {@code :authority}.
          */
-        AUTHORITY(":authority"),
+        AUTHORITY(":authority", REQUEST_PSEUDO_HEADER),
 
         /**
          * {@code :path}.
          */
-        PATH(":path"),
+        PATH(":path", REQUEST_PSEUDO_HEADER),
 
         /**
          * {@code :status}.
          */
-        STATUS(":status");
+        STATUS(":status", RESPONSE_PSEUDO_HEADER);
+
+        private static char PSEUDO_HEADER_PREFIX = ':';
 
         private final AsciiString value;
-        private static final CharSequenceMap<AsciiString> PSEUDO_HEADERS = new CharSequenceMap<AsciiString>();
+        private final HeaderType type;
+        private static final CharSequenceMap<PseudoHeaderName> PSEUDO_HEADERS = new CharSequenceMap<PseudoHeaderName>();
+
         static {
             for (PseudoHeaderName pseudoHeader : PseudoHeaderName.values()) {
-                PSEUDO_HEADERS.add(pseudoHeader.value(), AsciiString.EMPTY_STRING);
+                PSEUDO_HEADERS.add(pseudoHeader.value(), pseudoHeader);
             }
         }
 
-        PseudoHeaderName(String value) {
+        PseudoHeaderName(String value, HeaderType type) {
             this.value = AsciiString.cached(value);
+            this.type = type;
         }
 
         public AsciiString value() {
             // Return a slice so that the buffer gets its own reader index.
             return value;
         }
 
+        public HeaderType type() {
+            return type;
+        }
+
         /**
          * Indicates whether the given header name is a valid HTTP/2 pseudo header.
          */
         public static boolean isPseudoHeader(CharSequence header) {
             return PSEUDO_HEADERS.contains(header);
         }
+
+        /**
+         * Returns the {@link PseudoHeaderName} corresponding to the specified header name.
+         *
+         * @return corresponding {@link PseudoHeaderName} if any, {@code null} otherwise.
+         */
+        public static PseudoHeaderName getPseudoHeader(CharSequence header) {
+            return PSEUDO_HEADERS.get(header);
+        }
+
+        /**
+         * Indicates whether the specified header follows the pseudo-header format (begins with ':' character)
+         *
+         * @return {@code true} if the header follow the pseudo-header format
+         */
+        public static boolean hasPseudoHeaderFormat(CharSequence header) {
+            try {
+                if (header instanceof AsciiString) {
+                    return ((AsciiString) header).byteAt(0) == PSEUDO_HEADER_PREFIX;
```

Comment:
consider adding a new `private static final byte PSEUDO_HEADER_PREFIX_BYTE = (byte) ';';` and using it here to avoid any casting.

Reply:
- [x] add byte format for prefix